### ISSUE TITLE
Multi-arch images cannot be published to a repository with immutable tags

### ISSUE THREAD

**Issue by c0d1ngm0nk3y** [buildpacks/pack#2299]
### Description

When the repository does not allow to overwrite existing tags, it is not possible to publish an image for multiple architectures.

### Proposed solution

In case of of a multi-arch image, the actual manifest must not be written to the repository. Only the the layer. The manifest containing all digest will be created [later](https://github.com/buildpacks/pack/blob/c5fadcd444b6a82bfca0c0e5e501d5392a7cd8ef/pkg/client/package_buildpack.go#L94).

### Describe alternatives you've considered

The target repository needs to allow overwriting of published tags which might not always be desired / possible.

### Additional context
- [ ] This feature should be documented somewhere

Code in question: https://github.com/buildpacks/pack/blob/c5fadcd444b6a82bfca0c0e5e501d5392a7cd8ef/pkg/client/package_buildpack.go#L83-L99

The tag will be created for each architecture, just to be overwritten later with the combined manifest.

**Comment by jjbustamante** [comment]
Hi @loewenstein-sap, I am working on it, but I still don't have the fix

**Comment by jjbustamante** [comment]
@c0d1ngm0nk3y , @loewenstein-sap , @pbusko 

I've been trying to find a way to fix this without making many changes.

If I am not wrong the problem is the following: 
When we run `pack build my-repo/my-image:my-tag` for a multi-arch buildpack or builder, what pack is doing is:
- For each target
  - `pack` will create an OCI image and push it to the registry as  `my-repo/my-image:my-tag`, creating a manifest for each architecture.
- At the end, `pack` will create an Image Index to combine all the previous manifest created 

I think that process is not wrong, the problem appears when we uses the same image name `my-repo/my-image:my-tag` for each architecture, right? and when the registry doesn't allow overwriting an existent tag then an error is thrown?

*Proposal* 

what if `pack`, maybe with a `flag`, adds a suffix `os-architecture` at the end given `tag`, something like:
 - my-repo/my-image:my-tag-linux-amd64 (manifest-1)
 - my-repo/my-image:my-tag-linux-arm (manifest-2)
 - my-repo/my-image:my-tag (image index)

![ociimage (1)](https://github.com/user-attachments/assets/b1748881-2054-4204-b98a-0d86da54efd3)

Does this solution make sense to you? I think, I implemented it in this way during the PoC but I can't remember why I changed it, maybe because I tested against a mutable registry. 

image source: [go-containerregistry](https://github.com/google/go-containerregistry/blob/main/images/ociimage.jpeg)

**Comment by pbusko** [comment]
> I think that process is not wrong, the problem appears when we uses the same image name my-repo/my-image:my-tag for each architecture, right? and when the registry doesn't allow overwriting an existent tag then an error is thrown?

Correct. For comparison, if you run `crane copy <src> <dst>` for a multi-arch image, it writes a tag **only once** at the end. Ideally `pack` should follow the same principle and not create tags for individual images at all, but only during the `CreateManifest` call: https://github.com/buildpacks/pack/blob/2f1af856afc54fa1220be8f0c7d31c481e258f13/pkg/client/package_buildpack.go#L94

**Comment by jjbustamante** [comment]
Hi @c0d1ngm0nk3y @pbusko @loewenstein-sap 

Is it possible for you to test the [binaries](https://github.com/buildpacks/pack/actions/runs/12800612338) created for the branch?

**Comment by c0d1ngm0nk3y** [comment]
> Is it possible for you to test the [binaries](https://github.com/buildpacks/pack/actions/runs/12800612338) created for the branch?

With the provided binary, the `pack buildpack package --publish ... --append-image-name-suffix` is successful.

**Comment by jjbustamante** [comment]
> > Is it possible for you to test the [binaries](https://github.com/buildpacks/pack/actions/runs/12800612338) created for the branch?
> 
> With the provided binary, the `pack buildpack package --publish ... --append-image-name-suffix` is successful.

Thank you!!! @c0d1ngm0nk3y
